e2e/setup/e2e-setup.ts

```diff
@@ -0,0 +1,103 @@
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+ * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+ * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+import { config } from "dotenv-flow";
+
+config({
+  path: __dirname.concat("/../env/"),
+  // Disable warning messages in CI
+  silent: process.env.CI === "true",
+});
+
+const availableEnvironment = [
+  "ESS Dev-Next" as const,
+  "ESS Production" as const,
+  "ESS 1.1" as const,
+  "NSS" as const,
+];
+
+export type AvailableEnvironment = typeof availableEnvironment extends Array<
+  infer E
+>
+  ? E
+  : never;
+
+function isAvailableEnvironment(
+  environment: string | undefined
+): environment is AvailableEnvironment {
+  return availableEnvironment.includes(environment as AvailableEnvironment);
+}
+
+export interface TestingEnvironment {
+  clientId: string;
+  clientSecret: string;
+  environment: AvailableEnvironment;
+  // No environment-specific features at the moment
+  // feature: {}
+  idp: string;
+  pod: string;
+}
+
+function validateEnvVariables() {
+  let missing = false;
+  let missingMessage = "";
+
+  if (missing) {
+    throw new Error(`Environment variable missing: ${missingMessage}`);
+  }
+}
+
+export function getTestingEnvironment(): TestingEnvironment {
+  if (!isAvailableEnvironment(process.env.E2E_TEST_ENVIRONMENT)) {
+    throw new Error(`Unknown environment: ${process.env.E2E_TEST_ENVIRONMENT}`);
+  }
+  validateEnvVariables();
+  if (typeof process.env.E2E_TEST_POD !== "string") {
+    throw new Error("The environment variable E2E_TEST_POD is undefined.");
+  }
+  if (typeof process.env.E2E_TEST_IDP !== "string") {
+    throw new Error("The environment variable E2E_TEST_IDP is undefined.");
+  }
+  if (typeof process.env.E2E_TEST_CLIENT_ID !== "string") {
+    throw new Error(
+      "The environment variable E2E_TEST_CLIENT_ID is undefined."
+    );
+  }
+  if (typeof process.env.E2E_TEST_CLIENT_SECRET !== "string") {
+    throw new Error(
+      "The environment variable E2E_TEST_CLIENT_SECRET is undefined."
+    );
+  }
+
+  return {
+    pod: process.env.E2E_TEST_POD,
+    idp: process.env.E2E_TEST_IDP,
+    clientId: process.env.E2E_TEST_CLIENT_ID,
+    clientSecret: process.env.E2E_TEST_CLIENT_SECRET,
+    environment: process.env.E2E_TEST_ENVIRONMENT,
+    // feature: {}
+  };
+}
+
+export function formatForDisplay(iri: string): string {
+  // Trim `https://` from the start of these URLs,
+  // so that GitHub Actions doesn't replace them with *** in the logs.
+  return iri.replace(/^https?:\/\//, "");
+}
```

Comment:
I don't think you want to use that. Those are secrets, they should be replaced with *** in the logs.

The logs will show which environment the tests are running against. That's enough.

Reply:
Fair point, having the environment prevents from having to use this

Reply:
Done in 807b7318c71e16a7f36f1f20d1601dae8ac8a1bb